### ISSUE TITLE
Foreach loop adds blank line

### ISSUE THREAD

**Issue by ZakMiller** [nicoespeon/abracadabra#262]
## Describe the bug

When converting a for loop to a foreach loop, if there is a line directly above the line being tranformed then a blank line is added.

## How to reproduce
![2021-03-23 14 32 16](https://user-images.githubusercontent.com/18072671/112199842-f68a5f80-8be4-11eb-9355-2fcf34a9987d.gif)

## Expected behavior

The blank line should not be added.

## Additional information

- Version of the extension impacted: 5.0.1.

I discovered this issue while working on #92.

**Comment by nicoespeon** [comment]
Ah yeah, that's likely a side-effect from babel or recast (which we use to parse AST and generate code from it). Hopefully, that should be easy to override. I'll have a look.

Thanks for reporting and working on the for-loop refactoring. Much appreciated ❤️

**Comment by nicoespeon** [comment]
I had a quick look and it's trickier than expected. This is happening automatically when recast prints the modified AST. I didn't found a simple way to override this behavior.

Thus, I think I'll leverage the InMemory editor capabilities we've built to address this issue on our side directly. 

That should be robust enough: the transformed node start position should have the same `line` value. If it's not and that line was replaced with a blank line, we should be able to transform the code to remove the added line. All of this before we send the code to the editor, so it's transparent for VS Code (= faster + atomic update).
